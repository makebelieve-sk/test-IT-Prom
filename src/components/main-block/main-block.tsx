import React, { useState } from 'react';

import { mockList } from '../../mock-data';
import { TableComponent } from '../table-component/table-component';
import { DataType } from '../../types';

export const MainBlock: React.FC = () => {
    const [data, setData] = useState(null);
    let obj: DataType;

    const clickHandler = (el: DataType) => {
        obj = {
            header: el.header,
            body: el.body
        };
        setData(obj);
    };

    return (
        <div className="main-block">
            <div className="list-block">
                { mockList.map((el, i) => {
                    return <div key={`${el}-${i}`} className="button-in-list" onClick={clickHandler.bind(this, el)}>{ el.name }</div>
                }) }
            </div>
            { data ? <TableComponent data={data} setData={setData} /> : null}
        </div>
    )
};