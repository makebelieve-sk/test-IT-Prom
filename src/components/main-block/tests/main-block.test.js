import React from 'react';
import ReactDOM from 'react-dom'; 
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { MainBlock } from '../main-block';

Enzyme.configure({ adapter: new Adapter() });

it(`MainBlock correctly rendering`, () => {
    const tree = renderer  
        .create(<MainBlock />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

describe('MainBlock correctly rendering and functioning', () => {
    let wrapper;
  
    beforeEach(() => {
        wrapper = shallow(<MainBlock />);
    });

    it(`+++ MainBlock renders without crashing`, () => {
        const div = document.createElement('div');
      
        ReactDOM.render(<MainBlock />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('+++ render the connected MainBlock', () => {
        expect(wrapper.length).toEqual(1);
    });
});