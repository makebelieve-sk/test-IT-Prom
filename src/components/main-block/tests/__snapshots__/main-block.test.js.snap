// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBlock correctly rendering 1`] = `
<div
  className="main-block"
>
  <div
    className="list-block"
  >
    <div
      className="button-in-list"
      onClick={[Function]}
    >
      Профессии
    </div>
    <div
      className="button-in-list"
      onClick={[Function]}
    >
      Отделы
    </div>
    <div
      className="button-in-list"
      onClick={[Function]}
    >
      Сотрудники
    </div>
  </div>
</div>
`;
